python/llm/src/ipex_llm/transformers/speculative.py

```diff
@@ -510,7 +510,7 @@ def _crop_past_key_values(self, past_key_values, new_cache_size, _enable_ipex=Fa
                 for k, v in past_key_values
             ]
         elif self.config.model_type == "chatglm":
-            if self.config.num_layers == 40 and hasattr(self.config, 'rope_ratio'):
+            if self.config.num_layers in [28, 40] and hasattr(self.config, 'rope_ratio'):
```

Comment:
add comments (on which models these conditions are intended for)

Reply:
Thank you for the comment. Have updated :)